server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -283,12 +283,16 @@ private void serveWellKnown(ServletRequestDetails request) {
   void mutateRequest(RequestDetails requestDetails, AccessDecision accessDecision) {
     RequestMutation mutation =
         accessDecision.getRequestMutation(new RequestDetailsToReader(requestDetails));
-    if (mutation == null || CollectionUtils.isEmpty(mutation.getQueryParams())) {
+    if (mutation == null) {
       return;
     }
-
-    mutation
-        .getQueryParams()
-        .forEach((key, value) -> requestDetails.addParameter(key, value.toArray(new String[0])));
+    if (!CollectionUtils.isEmpty(mutation.getQueryParams())) {
+      mutation
+          .getQueryParams()
+          .forEach((key, value) -> requestDetails.addParameter(key, value.toArray(new String[0])));
+    }
+    if (mutation.getRequestContent().length > 0) {
+      requestDetails.setRequestContents(mutation.getRequestContent());
```

Comment:
Can you please add a unit-test that exercises this? For example something like [this](https://github.com/google/fhir-gateway/blob/f0788a647799d92ce8111782367281e6b859f503/server/src/test/java/com/google/fhir/gateway/BearerAuthorizationInterceptorTest.java#L275).

Also can you please fill-up the TESTED field of the PR description and describe how you have tested this change. Especially if your AccessChecker plugin is open-source, that would be great to point to that as well.